### PR TITLE
Test iSCSI target resources

### PR DESCRIPTION
**Pull request by lucaskanashiro**
Adding tests for `iSCSITarget` and `iSCSILogicalUnit` resources.

### CODE REVIEW

File: ha/virsh/tests/resource_iscsi_target_test.sh
Comment (athos-ribeiro) [inline ha/virsh/tests/resource_iscsi_target_test.sh:48]: I recently proposed a set of PRs to move away from performing these assertions without information for the OCI tests.

The reasoning behind it is that, shall the test fail, one cannot easily infer why it did so nor even speculate what is going on without at least opening the test scripts and/or trying to reproduce the failure.

Would you think adding more info here could help in this matter as well?

For instance, you could do something like `assertTrue "Could not infer ${LUN_RESOURCE_NAME} readiness from status output:\n${cluster_status}" $?`

**Comment by paride** [comment]
I hit a failure in Jenkins when testing with `UBUNTU_SERIES=impish`, however it seems related to the cluster setup and not to this specific test (see below). I retriggered it and it worked fine, while the Jammy job worked fine from the beginning.

At the moment we try to ensure the VMs are ready with a simple `sleep 180`, maybe that wasn't enough because of high load of the host machine? That logic is something to improve, but not a blocker for landing this .

I'm waiting for resolution of @athos-ribeiro's review comment before landing. The Jenkins side of this is ready.

Log excerpt:

```
+ get_network_data_nic2
++ virsh domifaddr ha-agent-virsh-impish-resource-iscsi-target-node01
++ grep ipv4
++ xargs
++ cut -d ' ' -f8
++ cut -d / -f1
error: Failed to query for interfaces addresses
error: invalid network pointer in virNetworkGetDHCPLeases
+ IP2_VM01=
error: Failed to query for interfaces addresses
error: invalid network pointer in virNetworkGetDHCPLeases
error: Failed to query for interfaces addresses
error: invalid network pointer in virNetworkGetDHCPLeases
error: Failed to query for interfaces addresses
error: invalid network pointer in virNetworkGetDHCPLeases
error: Failed to query for interfaces addresses
error: invalid network pointer in virNetworkGetDHCPLeases
error: Failed to query for interfaces addresses
error: invalid network pointer in virNetworkGetDHCPLeases
error: Failed to query for interfaces addresses
error: invalid network pointer in virNetworkGetDHCPLeases
ssh: Could not resolve hostname : Name or service not known
ASSERT:Unknown failure encountered running a test
```

File: ha/virsh/tests/resource_iscsi_target_test.sh
Comment (lucaskanashiro) [reply ha/virsh/tests/resource_iscsi_target_test.sh:48]: That's indeed something to we should improve. I'll apply those changes in this PR and update the other tests when I touch them again.

**Comment by lucaskanashiro** [comment]
@paride the agents that we are testing now can be tested only in Jammy, I'll not be backporting them to Impish.

**Comment by lucaskanashiro** [comment]
Now I believe it is good to go. Addressed @athos-ribeiro 's comment.

**Review by paride — APPROVED** [review]
LGTM, thanks! I'm re-running this in Jenkins out of the PR branch, if it succeeds I'll land it.

**Review by paride — CHANGES_REQUESTED** [review]
Fails for me both in Jenkins and locally (`TESTS=tests/resource_iscsi_target_test.sh ./run_tests.sh`)

```
ASSERT:
test_iscsi_target_info
o- iqn.2022-02.com.test:target ........................................................................................... [TPGs: 1]
    | o- lun0 ....................................................... [block/iscsi-lun0-rsc (/dev/mapper/mpatha) (default_tg_pt_gp)]
test_move_resource
INFO: Move constraint created for iscsi-target-rsc to ha-agent-virsh-jammy-resource-iscsi-target-node02
    * iscsi-target-rsc	(ocf:heartbeat:iSCSITarget):	 Started ha-agent-virsh-jammy-resource-iscsi-target-node02
    * iscsi-lun0-rsc	(ocf:heartbeat:iSCSILogicalUnit):	 Started ha-agent-virsh-jammy-resource-iscsi-target-node02
o- iqn.2022-02.com.test:target ........................................................................................... [TPGs: 1]
    | o- lun0 ....................................................... [block/iscsi-lun0-rsc (/dev/mapper/mpatha) (default_tg_pt_gp)]
```

**Comment by lucaskanashiro** [comment]
@paride it seems to be fixed now, could you try it again?

**Review by paride — APPROVED** [review]
All good now, thanks!

**Pull request merged**
